src/chapters/aspdotnet-model-classes/data-layer.rst

```diff
@@ -0,0 +1,284 @@
+.. _data-layer:
+
+Models and Data 
+===============
+
+In order to work with data, we need to add another element to our MVC application. Say for example,
+we want to do things like remove an event from our list. Well, if two events both have the same name, 
+how might we identify which of those items to delete? We can't yet. So we need to tweak how we store
+event data. 
+
+In ``CodingEvents``, we add a unique identifier field to ``Events`` to better handle and track distinct 
+``Event`` instances. We'll also create another model class called ``EventData`` to encapsulate data storage 
+and prepare ourselves for decoupling data from controller classes.
+
+Add a Unique Id - Video
+-----------------------
+
+.. TODO: Add create uid in model class video
+.. topics covered: create id field on event class
+
+YOUTUBE VIDEO HERE
+
+.. admonition:: Note
+
+   The starter code for this video is found at the `add-property branch <https://github.com/LaunchCodeEducation/CodingEventsDemo/tree/add-property>`__
+   of ``CodingEventsDemo``. The final code presented in this 
+   video is found on the `add-id branch <https://github.com/LaunchCodeEducation/CodingEventsDemo/tree/add-id>`__.
+   As always, code along to the videos on your own ``CodingEvents`` project.
+
+.. index:: ! uid, ! unique identifier
+
+Add a Unique Id - Text
+----------------------
+
+Identifying data by a user-defined string called ``name`` is not a sustainable or scalable method
+of handling data in most situations. Consider the address book example. How can
+we distinguish between two contact entries with the same name field? It is a frequent
+practice to add a **unique identifier** field (sometimes called, or even labelled, **uid**) to a class 
+responsible for modelling data. This ensures that our address book can contain two separate entries for 
+our contacts who have the same name as one another. 
+
+To accomplish the same data clarity with events, we'll add a few things to the event model class:
+
+#. A readonly ``id`` field.
+#. A static counter variable, ``nextId``.
+#. Additional constructor code that:
+   
+   a. Sets ``id`` to the ``nextId`` value.
+   b. Increments ``nextId``.
+
+The result in ``Models/Event.cs``:
+
+.. sourcecode:: c#
+   :linenos: 
+
+   namespace CodingEventsDemo.Models
+   {
+      public class Event
+      {
+         public string Name { get; set; }
+         public string Description { get; set; }
+
+         public int Id { get; }
+         static private int nextId = 1;
+
+         public Event(string name, string description)
+         {
+            Name = name;
+            Description = description;
+            Id = nextId;
+            nextId++;
+         }
+
+         public override string ToString()
+         {
+            return Name;
+         }
+      }
+   }
+
+.. index:: ! get-only auto-implemented property
```

Comment:
They have seen this syntax before in the context of ``readonly`` fields in Chapter 5.

Reply:
yes, i thought about that too. but Im also thinking the label and definition are good here since we're giving a more in-depth look at what is taking place with the backing field

Reply:
I agree! I would just add a reference back to the original context and an index entry for ``readonly``.